packages/blockchain-wallet-v4-frontend/src/hooks/useFetcher/useFetcher.tsx

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { FetcherHookData } from './useFetcher.types'
+
+const useFetcher = <T extends unknown = unknown, E extends object = object>(path: string) => {
+  const [state, setState] = useState<FetcherHookData<T, E>>({
+    data: undefined,
+    isFetching: false,
+    isLoading: false
+  })
+
+  const startFetch = useCallback(
+    async (path: string) => {
+      try {
+        setState((state) => ({
+          data: state.data,
+          error: undefined,
+          isFetching: true,
+          isLoading: state.data === undefined
+        }))
```

Comment:
I see a use case for a fetch that is non-remote based but if you need the states of `isFetching` `isLoading` and `error`, then you might as well return a `Remote`

Reply:
Yes, I could return a Remote type, but the remote type has an isNotAsked state that the fetch doesn't have because it will start fetching the data on the first render.
So, I thought it was not the right type to use there